pkg/daemon/controller/controller.go

```diff
@@ -488,9 +488,21 @@ func (sc *ServiceController) resolveNodeVariables(svc servicescm.Service) (map[s
 func (sc *ServiceController) resolvePowershellVariables(svc servicescm.Service) (map[string]string, error) {
 	vars := make(map[string]string)
 	for _, script := range svc.PowershellPreScripts {
+		// replaces the values in the PowershellPreScripts path with the node args
+		if len(script.NodeArgs) != 0 {
+			psNodeVars, err := sc.resolveNodeVariables(script.NodeArgs)
+			if err != nil {
+				return nil, err
+			}
+			pwshCommand := script.Path
+			for key, value := range psNodeVars {
+				pwshCommand = strings.ReplaceAll(pwshCommand, key, value)
+			}
+			script.Path = pwshCommand
+		}
 		out, err := sc.psCmdRunner.Run(script.Path)
 		if err != nil {
-			return nil, fmt.Errorf("could not resolve PowerShell variable %s: %w", script.VariableName, err)
+			return nil, fmt.Errorf("could not run PowerShell script %s: %w", script.Path, err)
```

Comment:
not sure how i feel about including the entire powershell script. might affect readability of the error msg. consider logging it instead

Reply:
Its not the entire powershell script. It is the path + args

Reply:
Fair enough, though the script "Path" is a bit of a misnomer since it can be an actual [in-line powershell](https://github.com/openshift/windows-machine-config-operator/blob/master/pkg/services/services.go#L221-L222) commands